### PR TITLE
[qt] Add Outdoors layer

### PR DESCRIPTION
**Pull request by Ferenc-**

### CODE REVIEW

**Review by pastk — APPROVED** [review]
A great change, many thanks!

**Comment by pastk** [comment]
It'be also great to have a Vehicle style button and remove the defunct Traffic.

File: qt/draw_widget.cpp [hunk 1]
```diff
@@ -781,6 +768,24 @@ void DrawWidget::RefreshDrawingRules()
   SetMapStyle(MapStyleDefaultLight);
 }
 
+void DrawWidget::SetMapStyleToDefault()
+{
+  auto style = m_framework.GetMapStyle();
```
Comment (biodranik) [inline qt/draw_widget.cpp:773]: nit here and below:
```suggestion
  auto const style = m_framework.GetMapStyle();
```

File: indexer/map_style.cpp [hunk 2]
```diff
@@ -53,3 +53,10 @@ std::string DebugPrint(MapStyle mapStyle)
 {
   return MapStyleToString(mapStyle);
 }
+
+
+bool MapStyleIsDark(MapStyle mapStyle)
+{
+  const std::unordered_set<MapStyle> darkStyles {MapStyle::MapStyleDefaultDark, MapStyle::MapStyleVehicleDark, MapStyle::MapStyleOutdoorsDark};
+  return darkStyles.contains(mapStyle);
+}
```
Comment (biodranik) [inline indexer/map_style.cpp:62]: There is no need to pessimize and allocate memory on each call. Looking up in 3 values in a sequential container is way faster than creating a hash and looking into it.
```suggestion
bool IsMapStyleDark(MapStyle mapStyle)
{
  using enum MapStyle;
  for (const auto darkStyle : {MapStyleDefaultDark, MapStyleVehicleDark, MapStyleOutdoorsDark})
  {
    if (mapStyle == darkStyle)
      return true;
  }
  return false;
}
```

**Review by biodranik — APPROVED** [review]
Thanks for the great change!

@pastk LGTM after fixing the lookup code.

File: qt/draw_widget.cpp (hunk 1 above)
Comment (Ferenc-) [reply qt/draw_widget.cpp:773]: Added to all 3 places.

File: indexer/map_style.cpp (hunk 2 above)
Comment (Ferenc-) [reply indexer/map_style.cpp:62]: Changed.

**Comment by biodranik** [comment]
No need to re-push. Checks can be rerun from the GitHub UI.

**Pull request merged**